src/varname.jl

```diff
@@ -575,12 +575,13 @@ julia> x = (a = [(b = rand(2), )], ); getlens(@varname(x.a[1].b[end], true))
     Using `begin` in an indexing expression to refer to the first index requires at least
     Julia 1.5.
 """
-macro varname(expr::Union{Expr,Symbol}, concretize::Bool=false)
+macro varname(expr::Union{Expr,Symbol}, concretize::Bool=Setfield.need_dynamic_lens(expr))
     return varname(expr, concretize)
 end
 
-varname(sym::Symbol, concretize=false) = :($(AbstractPPL.VarName){$(QuoteNode(sym))}())
-function varname(expr::Expr, concretize=false)
+varname(sym::Symbol) = :($(AbstractPPL.VarName){$(QuoteNode(sym))}())
```

Comment:
Presumably, these two `varname` are for backward compatibility. If so, maybe add a comment, so we don't forget.

Reply:
Nah, it has nothing to do with backwards compat. It's just that IMO this makes it clearer that `varname(::Symbol, ...)` does not at all depend on `concretize` vs. the previous impl which has `concretize=false` for `::Symbol`.

I can revert the change if you disagree:)

Reply:
Oh, that makes sense. I am happy to keep them but perhaps add a comment to clarify your point.